### PR TITLE
vcd_nsxt_ipsec_vpn_tunnel improvements

### LINKED ISSUE #926: vcd_nsxt_ipsec_vpn_tunnel ability to set remote id
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In VCD you can configure a remote ID for each IPSec VPN, it is possible to do this via GUI and API.
GUI:
image

API:
`{
  "id": "string",
  "name": "string",
  "description": "string",
  "enabled": true,
  "localEndpoint": {
    "localId": "string",
    "localAddress": "string",
    "localNetworks": [
      "string"
    ]
  },
  "remoteEndpoint": {
    "remoteId": "string",
    "remoteAddress": "string",
    "remoteNetworks": [
      "string"
    ]
  },
  "authenticationMode": "string",
  "preSharedKey": "string",
  "certificateRef": {
    "name": "string",
    "id": "string"
  },
  "caCertificateRef": {
    "name": "string",
    "id": "string"
  },
  "connectorInitiationMode": {},
  "securityType": "string",
  "logging": false,
  "version": {
    "version": 0
  }
}`

### New or Affected Resource(s)

* vcd_nsxt_ipsec_vpn_tunnel

### Terraform Code (how it would look like)
`resource "vcd_nsxt_ipsec_vpn_tunnel" "tunnel1" {
  org = "my-org"

  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id

  name        = "First"
  description = "testing tunnel"

  pre_shared_key = "my-presharaed-key"
  local_ip_address = tolist(data.vcd_nsxt_edgegateway.existing_gw.subnet)[0].primary_ip
  local_networks   = ["10.10.10.0/24", "30.30.30.0/28", "40.40.40.1/32"]
  remote_ip_address = "1.2.3.4"
  **remote_id = "5.6.7.8"**
  remote_networks   = ["192.168.1.0/24", "192.168.10.0/24", "192.168.20.0/28"]
}`

### PR DESCRIPTION
**Pull request by Didainius**
Closes #926 

This PR  adds three things:
* The option to explicitly set `remote_id` (as per request in #926)
* `authentication_mode` field that supports two auth modes `PSK` and `CERTIFICATE` (defaults to `PSK` - Pre Shared Key mode as it was the only one before)
* new fields `certificate_id` and `ca_certificate_id` for `CERTIFICATE` authentication mode 

#### Testing
* New root certificate was added (IPSec VPN tunnel tests need it) and existing certificates are replaced to match the root certificate

**Note** Test configurations need new field https://github.com/vmware/terraform-provider-vcd/pull/1010/files#diff-03843633ea9bfe994d5b8348b89a14a672f8e2096c68cc21c9d732b113984b6dR187

Tests worked on 10.3.0, 10.4.0, 10.4.1

### CODE REVIEW

File: vcd/config_test.go [hunk 1]
```diff
@@ -184,6 +184,7 @@ type TestConfig struct {
 		Certificate2Path           string `json:"certificate2Path,omitempty"`           // absolute path to pem file
 		Certificate2PrivateKeyPath string `json:"certificate2PrivateKeyPath,omitempty"` // absolute path to private key pem file
 		Certificate2Pass           string `json:"certificate2Pass,omitempty"`           // absolute path to pem file
+		RootCertificatePath        string `json:"rootCertificatePath,omitempty"`        // absolute path to pem file
```
Comment (dataclouder) [inline vcd/config_test.go:187]: We should also update `sample_vcd_test_config.json`

File: vcd/datasource_vcd_nsxt_ipsec_vpn_tunnel.go [hunk 2]
```diff
@@ -69,6 +84,11 @@ func datasourceVcdNsxtIpSecVpnTunnel() *schema.Resource {
 				Computed:    true,
 				Description: "Public IPv4 Address of the remote device terminating the VPN connection",
 			},
+			"remote_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Remote ID",
```
Comment (dataclouder) [inline vcd/datasource_vcd_nsxt_ipsec_vpn_tunnel.go:90]:
```suggestion
				Description: "ID of the remote peer site",
```
If I understand the docs correctly

File: vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go [hunk 3]
```diff
@@ -90,6 +108,12 @@ func resourceVcdNsxtIpSecVpnTunnel() *schema.Resource {
 				Required:    true,
 				Description: "Public IPv4 Address of the remote device terminating the VPN connection",
 			},
+			"remote_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Custom remote ID. 'remote_ip_address' is used by default",
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go:115]: Should it be "remote ID of the peer site"?

File: vcd/config_test.go (hunk 1 above)
Comment (Didainius) [reply vcd/config_test.go:187]: Added

File: vcd/datasource_vcd_nsxt_ipsec_vpn_tunnel.go (hunk 2 above)
Comment (Didainius) [reply vcd/datasource_vcd_nsxt_ipsec_vpn_tunnel.go]: Yes, this is a valid comment. changed.

File: vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go (hunk 3 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go]: Yes, this is a valid comment. changed.

**Review by dataclouder — APPROVED** [review]

**Review by lvirbalas — APPROVED** [review]
Thank you!

**Review by adambarreiro — APPROVED** [review]
LGTM!

**Review by adezxc — APPROVED** [review]

**Pull request merged**